File: example-site-interface/src/main/java/ai/core/example/site/api/ExampleAJAXWebService.java
package ai.core.example.site.api;

import ai.core.example.site.api.socialmedia.CreateEndUserSocialMediaAJAXRequest;
import ai.core.example.site.api.socialmedia.CreateEndUserSocialMediaAJAXResponse;
import ai.core.example.site.api.socialmedia.CreateSocialMediaAJAXRequest;
import ai.core.example.site.api.socialmedia.CreateSocialMediaAJAXResponse;
import ai.core.example.site.api.socialmedia.CreateSocialMediaIdeasAJAXResponse;
import ai.core.example.site.api.socialmedia.FillingImageAJAXRequest;
import ai.core.example.site.api.socialmedia.FillingImageAJAXResponse;
import ai.core.example.site.api.socialmedia.IpAdapterFaceIdImageAJAXRequest;
import ai.core.example.site.api.socialmedia.IpAdapterFaceIdImageAJAXResponse;
import ai.core.example.site.api.socialmedia.IpAdapterImageAJAXRequest;
import ai.core.example.site.api.socialmedia.IpAdapterImageAJAXResponse;
import ai.core.example.site.api.socialmedia.RelightingImageAJAXRequest;
import ai.core.example.site.api.socialmedia.RelightingImageAJAXResponse;
import ai.core.example.site.api.socialmedia.StyleShapeImageAJAXRequest;
import ai.core.example.site.api.socialmedia.StyleShapeImageAJAXResponse;
import core.framework.api.web.service.PUT;
import core.framework.api.web.service.Path;

/**
 * @author stephen
 */
public interface ExampleAJAXWebService {
    @PUT
    @Path("/ajax/example/social-media")
    CreateSocialMediaAJAXResponse createPost(CreateSocialMediaAJAXRequest request);

    @PUT
    @Path("/ajax/example/ideas")
    CreateSocialMediaIdeasAJAXResponse idea();

    @PUT
    @Path("/ajax/example/end-user/social-media")
    CreateEndUserSocialMediaAJAXResponse createEndUserPost(CreateEndUserSocialMediaAJAXRequest request);

    @PUT
    @Path("/ajax/example/end-user/ideas")
    CreateSocialMediaIdeasAJAXResponse endUserIdea();

    @PUT
    @Path("/ajax/example/relighting")
    RelightingImageAJAXResponse relighting(RelightingImageAJAXRequest request);

    @PUT
    @Path("/ajax/example/filling")
    FillingImageAJAXResponse filling(FillingImageAJAXRequest request);

    @PUT
    @Path("/ajax/example/ip-adapter")
    IpAdapterImageAJAXResponse ipAdapter(IpAdapterImageAJAXRequest request);

    @PUT
    @Path("/ajax/example/style-shape")
    StyleShapeImageAJAXResponse styleShape(StyleShapeImageAJAXRequest request);

    @PUT
    @Path("/ajax/example/face-id")
    IpAdapterFaceIdImageAJAXResponse ipAdapterFaceId(IpAdapterFaceIdImageAJAXRequest request);
}

File: example-site-frontend/src/service/ExampleAJAXWebService.ts
import {CreateSocialMediaIdeasAJAXResponse,CreateEndUserSocialMediaAJAXRequest,CreateEndUserSocialMediaAJAXResponse,IpAdapterFaceIdImageAJAXRequest,IpAdapterFaceIdImageAJAXResponse,FillingImageAJAXRequest,FillingImageAJAXResponse,GetGenmoAJAXResponse,IpAdapterImageAJAXRequest,IpAdapterImageAJAXResponse,RelightingImageAJAXRequest,RelightingImageAJAXResponse,CreateSocialMediaAJAXRequest,CreateSocialMediaAJAXResponse,StyleShapeImageAJAXRequest,StyleShapeImageAJAXResponse} from "type/api";
import {ajax} from "@wonder/core-fe";

export class ExampleAJAXWebService {
static endUserIdea(): Promise<CreateSocialMediaIdeasAJAXResponse>{
return ajax("PUT", "/ajax/example/end-user/ideas", {}, null);
}
static createEndUserPost(request:CreateEndUserSocialMediaAJAXRequest): Promise<CreateEndUserSocialMediaAJAXResponse>{
return ajax("PUT", "/ajax/example/end-user/social-media", {}, request);
}
static ipAdapterFaceId(request:IpAdapterFaceIdImageAJAXRequest): Promise<IpAdapterFaceIdImageAJAXResponse>{
return ajax("PUT", "/ajax/example/face-id", {}, request);
}
static filling(request:FillingImageAJAXRequest): Promise<FillingImageAJAXResponse>{
return ajax("PUT", "/ajax/example/filling", {}, request);
}
static getGenmo(id:string): Promise<GetGenmoAJAXResponse>{
return ajax("GET", "/ajax/example/genmo/:id", {id}, null);
}
static idea(): Promise<CreateSocialMediaIdeasAJAXResponse>{
return ajax("PUT", "/ajax/example/ideas", {}, null);
}
static ipAdapter(request:IpAdapterImageAJAXRequest): Promise<IpAdapterImageAJAXResponse>{
return ajax("PUT", "/ajax/example/ip-adapter", {}, request);
}
static relighting(request:RelightingImageAJAXRequest): Promise<RelightingImageAJAXResponse>{
return ajax("PUT", "/ajax/example/relighting", {}, request);
}
static createPost(request:CreateSocialMediaAJAXRequest): Promise<CreateSocialMediaAJAXResponse>{
return ajax("PUT", "/ajax/example/social-media", {}, request);
}
static styleShape(request:StyleShapeImageAJAXRequest): Promise<StyleShapeImageAJAXResponse>{
return ajax("PUT", "/ajax/example/style-shape", {}, request);
}
}